client/src/nonblocking/pubsub_client.rs

```diff
@@ -1,3 +1,168 @@
+//! A client for subscribing to messages from the RPC server.
+//!
+//! The [`PubsubClient`] implements [Solana WebSocket event
+//! subscriptions][spec].
+//!
+//! [spec]: https://docs.solana.com/developing/clients/jsonrpc-api#subscription-websocket
+//!
+//! This is a nonblocking (async) API. For a blocking API use the synchronous
+//! client in [`crate::pubsub_client`].
+//!
+//! A single `PubsubClient` client may be used to subscribe to many events via
+//! subscription methods like [`PubsubClient::account_subscribe`]. These methods
+//! return a [`PubsubClientResult`] of a pair, the first element being a
+//! [`BoxStream`] of subscription-specific [`RpcResponse`]s, the second being an
+//! unsubscribe closure, an asynchronous function that can be called and
+//! `await`ed to unsubscribe.
+//!
+//! Note that `BoxStream` contains an immutable reference to the `PubsubClient`
+//! that created it. This makes `BoxStream` not `Send`, forcing it to stay in
+//! the same task as its `PubsubClient`. `PubsubClient` though is `Send` and
+//! `Sync`, and can be shared between tasks by putting it in an `Arc`. Thus
+//! one viable pattern to creating multiple subscriptions is:
+//!
+//! - create an `Arc<PubsubClient>`
+//! - spawn one task for each subscription, sharing the `PubsubClient`.
+//! - in each task:
+//!   - create a subscription
+//!   - send the `UnsubscribeFn` to another task to handle shutdown
+//!   - loop while receiving messages from the subscription
+//!
+//! This pattern is illustrated in the example below.
+//!
+//! By default the [`PubsubClient::block_subscribe`] and
+//! [`PubsubClient::vote_subscribe`] events are disabled. They can be enabled by
```

Comment:
Should these link to the rpc_pubsub definitions instead of the client ones?
Also maybe:
```suggestion
//! By default the [`PubsubClient::block_subscribe`] and
//! [`PubsubClient::vote_subscribe`] events are disabled on RPC nodes. They can be enabled by
```

Reply:
I'm still looking into this suggestion.